enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/OnboardingProcessRepository.java

```diff
@@ -40,72 +42,133 @@
 @Repository
 public interface OnboardingProcessRepository extends CrudRepository<OnboardingProcessEntity, String> {
 
+    /**
+     * Find an onboarding process using process ID. Lock this process using PESSIMISTIC_WRITE lock
+     * until the end of the transaction.
+     *
+     * @param processId Onboarding process identifier.
+     * @return Optional onboarding process.
+     */
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
+    @Query("SELECT p FROM OnboardingProcessEntity p WHERE p.id = :processId")
+    Optional<OnboardingProcessEntity> findProcessByIdWithLock(String processId);
+
+    /**
+     * Find an existing process using identification data and status. Lock this process using PESSIMISTIC_WRITE lock
+     * until the end of the transaction.
+     *
+     * @param identificationData Identification data.
+     * @param status Process status.
+     * @return Optional onboarding process.
+     */
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
     @Query("SELECT p FROM OnboardingProcessEntity p WHERE p.status = :status AND p.identificationData = :identificationData")
-    Optional<OnboardingProcessEntity> findExistingProcessByIdentificationData(String identificationData, OnboardingStatus status);
+    Optional<OnboardingProcessEntity> findExistingProcessWithLock(String identificationData, OnboardingStatus status);
```

Comment:
Just not sure about naming (I know, it's not the crucial part of this PR) , what about

- `findProcessByIdWithLock` -> `findByIdWithLock` (we are in process repository, so process is obvious)
- `findExistingProcessWithLock` -> `findByIdentificationDataWithLock` (it's obvious that we are looking for an existing process)